src/posix/platform/radio.cpp

```diff
@@ -87,15 +87,62 @@ Radio::Radio(const char *aUrl)
 
 void Radio::Init(void)
 {
-    bool        resetRadio             = (mRadioUrl.GetValue("no-reset") == nullptr);
-    bool        restoreDataset         = (mRadioUrl.GetValue("ncp-dataset") != nullptr);
-    bool        skipCompatibilityCheck = (mRadioUrl.GetValue("skip-rcp-compatibility-check") != nullptr);
-    const char *parameterValue;
-    const char *region;
+    bool         resetRadio             = (mRadioUrl.GetValue("no-reset") == nullptr);
+    bool         restoreDataset         = (mRadioUrl.GetValue("ncp-dataset") != nullptr);
+    bool         skipCompatibilityCheck = (mRadioUrl.GetValue("skip-rcp-compatibility-check") != nullptr);
+    spinel_iid_t iid                    = SPINEL_HEADER_IID_0;
+    spinel_iid_t iidList[SPINEL_HEADER_IID_MAX + 1];
+    const char  *iidString     = (mRadioUrl.GetValue("iid"));
+    const char  *iidListString = (mRadioUrl.GetValue("iid-list"));
+    const char  *parameterValue;
+    const char  *region;
 #if OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE
     const char *maxPowerTable;
 #endif
 
+#if OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
+    VerifyOrDie(iidString != nullptr, OT_EXIT_INVALID_ARGUMENTS);
+    iid = static_cast<spinel_iid_t>(atoi(iidString));
+#else
+    VerifyOrDie(iidString == nullptr, OT_EXIT_INVALID_ARGUMENTS);
+#endif
+
+#if OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
+    // Initialize iidList with out-of-range iid values.
+    memset(iidList, SPINEL_HEADER_IID_MAX + 1, sizeof(iidList));
+
+    // Add IID of the host application to the list of IIDs to allow spinel frames.
+    iidList[0] = iid;
+
+    if (iidListString != nullptr)
+    {
+        // Convert string to an array of integers.
+        // Integer i is for traverse the iidListString.
+        // Integer j is for iidList array offset location.
+        for (uint8_t i = 0, j = 1; iidListString[i] != '\0' && j <= SPINEL_HEADER_IID_MAX; i++)
```

Comment:
`j` starts from `1` because the first entry in iid-list will always be the host application/process iid (see line 115). I will also add a comment for better code readability.